src/propslib/DelegationConfig.hpp

```diff
@@ -18,83 +18,104 @@ Author(s):
+        }
+    };
+
     struct PkgVersion
     {
-        unsigned short major;
-        unsigned short minor;
-        unsigned short build;
-        unsigned short revision;
+        unsigned short major = 0;
+        unsigned short minor = 0;
+        unsigned short build = 0;
+        unsigned short revision = 0;
 
-        bool operator==(const PkgVersion& other) const
+        constexpr bool operator==(const PkgVersion& other) const noexcept
         {
-            return major == other.major &&
-                   minor == other.minor &&
-                   build == other.build &&
-                   revision == other.revision;
+            static_assert(std::has_unique_object_representations_v<PkgVersion>);
+            return __builtin_memcmp(this, &other, sizeof(*this)) == 0;
         }
     };
 
-    struct DelegationBase
+    struct PackageInfo
     {
-        CLSID clsid;
         std::wstring name;
         std::wstring author;
         std::wstring pfn;
         std::wstring logo;
         PkgVersion version;
 
-        bool IsFromSamePackage(const DelegationBase& other) const
+        bool IsFromSamePackage(const PackageInfo& other) const noexcept
         {
             return name == other.name &&
                    author == other.author &&
                    pfn == other.pfn &&
                    version == other.version;
         }
-
-        bool operator==(const DelegationBase& other) const
-        {
-            return clsid == other.clsid &&
-                   name == other.name &&
-                   author == other.author &&
-                   pfn == other.pfn &&
-                   version == other.version;
-        }
-    };
-
-    struct DelegationConsole : public DelegationBase
-    {
     };
 
-    struct DelegationTerminal : public DelegationBase
+    struct DelegationBase
     {
+        CLSID clsid{};
+        PackageInfo info;
     };
 
     struct DelegationPackage
     {
-        DelegationConsole console;
-        DelegationTerminal terminal;
+        DelegationPair pair;
+        PackageInfo info;
 
-        bool operator==(const DelegationPackage& other) const
+        bool operator==(const DelegationPackage& other) const noexcept
         {
-            return console == other.console &&
-                   terminal == other.terminal;
+            return pair == other.pair;
         }
     };
 
     [[nodiscard]] static HRESULT s_GetAvailablePackages(std::vector<DelegationPackage>& packages, DelegationPackage& def) noexcept;
 
     [[nodiscard]] static HRESULT s_SetDefaultByPackage(const DelegationPackage& pkg, const bool useRegExe = false) noexcept;
 
-    [[nodiscard]] static HRESULT s_GetDefaultConsoleId(IID& iid) noexcept;
-    [[nodiscard]] static HRESULT s_GetDefaultTerminalId(IID& iid) noexcept;
+    static constexpr CLSID CLSID_Default{};
+    static constexpr CLSID CLSID_Conhost{ 0xb23d10c0, 0xe52e, 0x411e, { 0x9d, 0x5b, 0xc0, 0x9f, 0xdf, 0x70, 0x9c, 0x7d } };

```

Comment:
What happens when a user installs Terminal 1.15 on a Windows 11 machine that doesn't know about the new guid that means "conhost"?

Reply:
We can't use velocity, so I'd solve this with a version check in WT potentially?